components/automate-ui/src/app/services/projects-filter/projects-filter.effects.ts

```diff
@@ -27,11 +27,10 @@ export class ProjectsFilterEffects {
       return this.projectsFilter.fetchOptions().pipe(
         map((fetched: ProjectsFilterOption[]) => {
           const restored = this.projectsFilter.restoreOptions() || [];
-          const loaded = fetched.map(fetchedOpt => {
-            const restoredOpt = restored.filter(opt => opt.value === fetchedOpt.value)[0];
-            return restoredOpt ? { ...fetchedOpt, checked: restoredOpt.checked } : fetchedOpt;
+          return new LoadOptionsSuccess(<ProjectsFilterOptionTuple>{
+            fetched: fetched,
+            restored: restored
```

Comment:
Moved this logic into the reducer to (a) co-locate with the sorting of options and (b) to be able to readily add tests for it.